src/Strategy/Gradle/ResolutionApi.hs

```diff
@@ -0,0 +1,155 @@
+  insertEnvironment,
+ )
+import Effect.Grapher (LabeledGrapher, deep, direct, edge, label, run, withLabeling)
+import GHC.Generics (Generic)
+import Graphing (Graphing)
+import Strategy.Gradle.Common (
+  ConfigName (unConfigName),
+  GradleLabel (Env),
+  ProjectName (..),
+  configNameToLabel,
+  getLinesWithPrefix,
+  packagePathsWithJson,
+ )
+
+data ResolvedProject = ResolvedProject
+  { resolvedProjectName :: ProjectName
+  , resolvedProjectConfigurations :: [ResolvedConfiguration]
+  }
+  deriving (Show, Eq, Ord, Generic, FromJSON)
+
+data ResolvedConfiguration = ResolvedConfiguration
+  { resolvedConfigurationName :: ConfigName
+  , resolvedConfigurationDirectComponents :: [ResolvedDependency]
+  , resolvedConfigurationDependencies :: [ResolvedComponent]
+  }
+  deriving (Show, Eq, Ord, Generic, FromJSON)
+
+data ResolvedComponent = ResolvedComponent
+  { resolvedComponentNode :: ResolvedDependency
+  , resolvedComponentOutgoing :: [ResolvedDependency]
+  }
+  deriving (Show, Eq, Ord, Generic, FromJSON)
+
+data ResolvedDependency
+  = ProjectDependency Text
+  | PackageDependency Text Text
+  deriving (Show, Eq, Ord)
+
+instance FromJSON ResolvedDependency where
+  parseJSON = withObject "ResolvedDependency" $ \obj -> do
+    ty <- obj .: "type" :: Parser Text
+    case ty of
+      "project" -> ProjectDependency <$> obj .: "name"
+      "package" ->
+        PackageDependency
+          <$> obj .: "name"
+          <*> obj .: "version"
+      _ -> unexpected (String ty)
+
+parseResolutionApiJsonDeps :: Text -> [ResolvedProject]
+parseResolutionApiJsonDeps text = packagePathsWithDecoded
+  where
+    jsonDepsLines :: [Text]
+    jsonDepsLines = getLinesWithPrefix text "RESOLUTIONAPI_JSONDEPS_"
+
+    packagePathsWithDecoded :: [ResolvedProject]
+    packagePathsWithDecoded = mapMaybe (\(_, outJson) -> decodeStrict $ encodeUtf8 outJson) (packagePathsWithJson jsonDepsLines)
+
+-- | Builds graph using gradle's resolution api.
+-- Reference: https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/result/ResolutionResult.html
+buildGraph :: [ResolvedProject] -> Set.Set ConfigName -> Graphing Dependency
+buildGraph projects onlyConfigs = run . withLabeling toDependency $ mapM_ addConfig applicableConfigurations
+  where
+    applicableConfigurations :: [ResolvedConfiguration]
+    applicableConfigurations = filter (isConfigIncluded . resolvedConfigurationName) configurations
+
+    configurations :: [ResolvedConfiguration]
+    configurations = concatMap resolvedProjectConfigurations projects
+
+    isConfigIncluded :: ConfigName -> Bool
+    isConfigIncluded c = (Set.null onlyConfigs) || Set.member c onlyConfigs
+
+    addConfig :: Has (LabeledGrapher ResolvedDependency GradleLabel) sig m => ResolvedConfiguration -> m ()
+    addConfig resolvedConfig = do
+      let configLabel = toGradleLabel $ resolvedConfigurationName resolvedConfig
+
+      forM_ (resolvedConfigurationDirectComponents resolvedConfig) $ \directDep -> do
```

Comment:
[tiniest nit] prefer `Data.Foldable.for_` over `forM_` (the latter is semantically an alias to the former):

```hs
for_ :: (Foldable t, Applicative f) => t a -> (a -> f b) -> f ()
for_ = flip traverse_

forM_ :: (Foldable t, Monad m) => t a -> (a -> m b) -> m ()
forM_ = for_
```

might be a nit we can add to our hlint config 🤔